website/content/docs/concepts/plugins/storage/host-volumes.mdx

```diff
@@ -0,0 +1,387 @@
+      mkfs.ext4 "$path.ext4" 1>&2
+    fi
+    if ! is_mounted "$path"; then
+      mount "$path.ext4" "$path"
+    fi
+}
+
+delete_volume() {
+  local path="$1"
+  is_mounted "$path" && umount "$path"
+  rm -rf "$path"
+  rm -f "$path.ext4"
+}
+
+case "$1" in
+  "fingerprint")
+    printf '{"version": "%s"}' "$version"
+    ;;
+  "create")
+    path="$(get_path)"
+    stderr "creating volume at $path"
+    create_volume "$path" "$DHV_CAPACITY_MIN_BYTES"
+    # output what Nomad expects
+    bytes="$(stat --format='%s' "$path.ext4")"
+    printf '{"path": "%s", "bytes": %s}' "$path" "$bytes"
+    ;;
+  "delete")
+    path="$(get_path)"
+    stderr "deleting volume at $path"
+    delete_volume "$path" ;;
+  *)
+    echo "unknown operation: $1"
+    exit 1 ;;
+esac
+```
+
+</Tab>
+</Tabs>
+
+## Specification
+
+A plugin is registered with Nomad if it:
+
+* Is an executable file (a script or binary)
+* Is located in the [`client.host_volume_plugin_dir`][plugin_dir] directory
+* Responds appropriately to a `fingerprint` call
+
+### Operations
+
+To fully manage the lifecycle of a volume, plugins must fulfill all of the
+following operations:
+
+* [fingerprint](#fingerprint)
+* [create](#create)
+* [delete](#delete)
+
+Nomad passes the operation as the first positional argument to the plugin.
+That and other information are passed as environment variables. Environment
+variables are prefixed with `"DHV_"` (i.e. Dynamic Host Volume).
+
+Some variables may be required for certain plugins (namely, `DHV_CAPACITY_*`
+for plugins that can restrict size). Most are for plugin author convenience.
+
+#### fingerprint
+<blockquote style={{borderLeft: "solid 1px #00ca8e"}}>
+
+Nomad calls `fingerprint` when the client agent starts (or is reloaded with a
+SIGHUP) to discover valid plugins. The returned "version" is used to register
+the plugin on the Nomad node for volume scheduling.
+
+**CLI arguments:** `$1=fingerprint`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=fingerprint
+```
+
+**Expected stdout:**
+
+```
+{"version": "0.0.1"}
+```
+
+**Requirements:**
+
+* Must complete within 5 seconds, or Nomad will kill it
+  * should be much faster, as no actual work should be done
+* "version" value must be valid per the [hashicorp/go-version][go-version]
+  golang package
+
+</blockquote>
+
+#### create
+<blockquote style={{borderLeft: "solid 1px #00ca8e"}}>
+
+Nomad calls `create` when you run [`nomad volume create`][cli-create] CLI or
+use the [create API][api-create]).
+
+You can run `create` again for the same volume if you include the volume
+`id` in the volume specification.
+
+Nomad also calls `create` when the agent starts, for each volume that was
+previously created on the node, so plugins can ensure the volumes are available
+after an agent restart or host reboot.
+
+**CLI Arguments:** `$1=create`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=create
+DHV_HOST_PATH=/suggested/path/to/place/volume
```

Comment:
Do we want to define this further? i.e. is this path the path of the directory where the plugin should create the volume, or the path of the volume itself?

Reply:
in `main` currently it is the latter, but we're changing it to be the former, i.e. it will be a directory wherein the plugin should create a new volume directory (or mount, or what have you)

which of these would be clearer?

A) different wording in the example path

```suggestion
DHV_HOST_PATH=/put/your/volume/in/this/directory
```

B) use the `{}` convention

```suggestion
DHV_HOST_PATH={directory to put the volume in}
```

C) a more realistic example

```suggestion
DHV_HOST_PATH=/opt/nomad/data/host_volumes
```

D) something else?

Reply:
I like B here. That way we're not specifying anything that we don't intend to (ex. a relationship between the host path and the datadir).